pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,829 @@
+        if invert is not None:
+            self.invert = invert
+        if offset is not None:
+            self.offset = offset
+        if skew_factor is not None:
+            self.skew_factor = skew_factor
+        if probe_attenuation is not None:
+            self.probe_attenuation = probe_attenuation
+        if scale is not None:
+            self.scale = scale
+        if unit is not None:
+            self.unit = unit
+        if trigger_coupling is not None:
+            self.trigger_coupling = trigger_coupling
+        if trigger_level is not None:
+            self.trigger_level = trigger_level
+        if trigger_level2 is not None:
+            self.trigger_level2 = trigger_level2
+        if trigger_slope is not None:
+            self.trigger_slope = trigger_slope
+
+    @property
+    def current_configuration(self):
+        """ Read channel configuration as a dict containing the following keys:
+            - "channel": channel number (int)
+            - "attenuation": probe attenuation (float)
+            - "bandwidth_limit": bandwidth limiting enabled (bool)
+            - "coupling": "AC 1M", "DC 1M", "GND" coupling (str)
+            - "offset": vertical offset (float)
+            - "skew_factor": channel-tochannel skew factor (float)
+            - "display": currently displayed (bool)
+            - "unit": "A" or "V" units (str)
+            - "volts_div": vertical divisions (float)
+            - "inverted": inverted (bool)
+            - "trigger_coupling": trigger coupling can be "dc" "ac" "highpass" "lowpass" (str)
+            - "trigger_level": trigger level (float)
+            - "trigger_level2": trigger lower level for SLEW or RUNT trigger (float)
+            - "trigger_slope": trigger slope can be "negative" "positive" "window" (str)
+        """
+
+        ch_setup_raw = {
+            "channel": self.number,
+            "attenuation": self.probe_attenuation,
+            "bandwidth_limit": self.bwlimit,
+            "coupling": self.coupling,
+            "offset": self.offset,
+            "skew_factor": self.skew_factor,
+            "display": self.display,
+            "unit": self.unit,
+            "volts_div": self.scale,
+            "inverted": self.invert,
+            "trigger_coupling": self.trigger_coupling,
+            "trigger_level": self.trigger_level,
+            "trigger_level2": self.trigger_level2,
+            "trigger_slope": self.trigger_slope
+        }
+
+        # Convert values to specific type
+        to_str = ["coupling", "unit", "trigger_coupling", "trigger_slope"]
+        to_bool = ["bandwidth_limit", "display", "inverted"]
+        to_float = ["attenuation", "offset", "skew_factor", "volts_div", "trigger_level", "trigger_level2"]
+        to_int = ["channel"]
+        ch_setup = self.instrument.sanitize_dictionary(ch_setup_raw, to_bool=to_bool, to_float=to_float,
+                                                       to_int=to_int, to_str=to_str)
```

Comment:
I don'T understand this part -- your properties should already be defined to accept/return "sane" values (via the `map` or `cast` arguments of `control()` etc), there should be no need to "sanitize". 
Also, you are doing work that is already done AFAICT, e.g. `bwlimit` already returns a bool, yet you "sanitize" it again?

Reply:
I completely agree with you and I will refactor the code accordingly. The reason I am doing that is because that is more or less how it is done in the Keysight DSOX1102G driver.

Reply:
Refactoring is done. Now all properties accept and return the correct type.